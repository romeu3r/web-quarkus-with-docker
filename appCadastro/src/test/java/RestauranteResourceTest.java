import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.Test;

import javax.ws.rs.core.Response;

import static io.restassured.RestAssured.given;


@DBRider
//@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurante-senario1.yml")
    public void testBuscarRestaurantes() {
        String resultado = given()
                .when().get("/restaurantes")
                .then()
                .statusCode(Response.Status.OK.getStatusCode())
                .extract().asString();
        Approvals.verifyJson(resultado);
    }
}
