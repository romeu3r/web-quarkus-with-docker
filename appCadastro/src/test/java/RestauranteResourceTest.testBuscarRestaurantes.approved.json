[
  {
    "dataAtualizacao": "2022-06-01T13:16:13.234Z[UTC]",
    "dataCriacao": "2022-06-01T13:16:13.234Z[UTC]",
    "id": 3,
    "nome": "WIllian"
  },
  {
    "cnpj": "string",
    "dataAtualizacao": "2022-06-01T13:16:46.056Z[UTC]",
    "dataCriacao": "2022-06-01T13:13:42.133Z[UTC]",
    "id": 1,
    "nome": "WIllian_N",
    "proprietario": "string"
  },
  {
    "cnpj": "string",
    "dataAtualizacao": "2022-06-01T13:17:56.534Z[UTC]",
    "dataCriacao": "2022-06-01T13:13:45.022Z[UTC]",
    "id": 2,
    "nome": "WIllian_N2000",
    "proprietario": "string"
  },
  {
    "cnpj": "string",
    "dataAtualizacao": "2022-06-01T13:35:20.791Z[UTC]",
    "dataCriacao": "2022-06-01T13:35:20.791Z[UTC]",
    "id": 5,
    "nome": "BNovoB",
    "proprietario": "string"
  },
  {
    "cnpj": "string",
    "dataAtualizacao": "2022-06-01T15:42:46.304Z[UTC]",
    "dataCriacao": "2022-06-01T15:42:46.305Z[UTC]",
    "id": 6,
    "nome": "BNovoB",
    "proprietario": "string"
  },
  {
    "cnpj": "string",
    "dataAtualizacao": "2022-06-01T15:42:53.046Z[UTC]",
    "dataCriacao": "2022-06-01T15:42:53.046Z[UTC]",
    "id": 7,
    "nome": "Xnovooovn",
    "proprietario": "string"
  },
  {
    "cnpj": "string",
    "dataAtualizacao": "2022-06-01T16:09:46.547Z[UTC]",
    "dataCriacao": "2022-06-01T16:09:46.547Z[UTC]",
    "id": 8,
    "nome": "Betaaa",
    "proprietario": "string"
  }
]